kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/apps/v1/ReplicaSetOperationsImpl.java

```diff
@@ -78,23 +77,57 @@ public ReplicaSet updateImage(String image) {
     ReplicaSet oldRC = get();
 
     if (oldRC == null) {
-      throw new KubernetesClientException("Existing replica set doesn't exist");
+      throw new KubernetesClientException("Existing replication controller doesn't exist");
     }
     if (oldRC.getSpec().getTemplate().getSpec().getContainers().size() > 1) {
       throw new KubernetesClientException("Image update is not supported for multicontainer pods");
     }
-    if (oldRC.getSpec().getTemplate().getSpec().getContainers().size() == 0) {
+    if (oldRC.getSpec().getTemplate().getSpec().getContainers().isEmpty()) {
       throw new KubernetesClientException("Pod has no containers!");
     }
 
-    Container updatedContainer = new ContainerBuilder(oldRC.getSpec().getTemplate().getSpec().getContainers().iterator().next()).withImage(image).build();
+    Container container = oldRC.getSpec().getTemplate().getSpec().getContainers().get(0);
```

Comment:
nit: when retrieving the first item of a collection, in general it's better to do
```suggestion
    Container container = oldRC.getSpec().getTemplate().getSpec().getContainers().iterator().next();
```
This won't be affected by performance issues for specific collection implementations (imagine containers was some sort of LinkedList where access by index was heavily penalized)